h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -41,43 +41,101 @@ class SearchBarController extends Controller {
       }.bind(this), 0);
     };
 
-    var getPreviousSiblingElement = element => {
+    var isVisible = element => {
+      return element &&
+        (element.nodeType !== 1 ||
+          !element.classList ||
+          element.classList.contains('is-hidden'));
+    };
+
+    var getPreviousVisibleSiblingElement = element => {
       if (!element) {
         return null;
       }
 
       do {
         element = element.previousSibling;
-      } while (element && element.nodeType != 1);
-
+      } while (isVisible(element));
       return element;
     };
 
-    var getNextSiblingElement = element => {
+    var getNextVisibleSiblingElement = element => {
       if (!element) {
         return null;
       }
 
       do {
         element = element.nextSibling;
-      } while (element && element.nodeType != 1);
+      } while (isVisible(element));
 
       return element;
     };
 
+    var unHideDropdownItems = () => {
+      // Reset dropdown items to not hidden
+      this._dropdownItems.forEach(function(dropdownItem) {
+        dropdownItem.classList.remove('is-hidden');
+      });
+    };
+
     var closeDropdown = () => {
       clearActiveDropdownItem();
+      unHideDropdownItems();
       this.setState({open: false});
       this._input.removeEventListener('keydown', setupListenerKeys,
         true /*capture*/);
     };
 
     var openDropdown = () => {
+      clearActiveDropdownItem();
+
       this.setState({open: true});
+
       this._input.addEventListener('keydown', setupListenerKeys,
         true /*capture*/);
     };
 
+    var getVisibleDropdownItems = () => {
+      return this._dropdown.querySelectorAll('li:not(.is-hidden)');
+    };
+
+    var setVisibleDropdownItems = word => {
+      // Show items that match the word and ones that don't
+      this._dropdownItems.forEach(function(dropdownItem) {
+        var dropdownItemTitle =
+          dropdownItem.querySelector('[data-ref="searchBarDropdownItemTitle"]').
+            innerHTML.trim();
+        if (dropdownItemTitle.indexOf(word) < 0) {
+          dropdownItem.classList.add('is-hidden');
+        } else {
+          dropdownItem.classList.remove('is-hidden');
+        }
+      });
+    };
+
+    var maybeOpenOrCloseDropdown = () => {
+      var word = this._input.value.trim();
+      var shouldOpenDropdown = true;
+
+      setVisibleDropdownItems(word);
```

Comment:
I think it should be up to `handleInputOnInput()` to call `setVisibleDropdownItems()` and then call `maybeOpenOrCloseDropdown()`. That way calling `maybeOpenOrCloseDropdown()` does only one thing - either opens or closes the dropdown. Rather than updating the visibility of the dropdown items and then opening or closing the dropdown.

I _think_ `handleFocusinOnInput()` should only need to call `maybeOpenOrCloseDropdown()`, and not `setVisibleDropdownItems()`? It doesn't need to update the visibility.

You'll end up with `this._input.value.trim()` repeated twice in `setVisibleDropdownItems()` and then in `maybeOpenOrCloseDropdown()`, so it might be worth moving that into a helper.